src/shogun/io/SGIO.h

```diff
@@ -301,32 +301,35 @@ namespace shogun
 		template <typename... Args>
 		void SGIO::message(
 		    EMessageType prio, const SourceLocation& loc, const char* format,
-		    const Args&... args) const
+		    Args&&... args) const
 		{
 			if (should_log(prio))
 			{
 				fmt::memory_buffer msg;
-				fmt::format_to(msg, format, args...);
+				fmt::format_to(msg, format, std::forward<Args>(args)...);
 				message_(prio, loc, fmt::string_view(msg.data(), msg.size()));
 			}
 		}
 
 		template <typename... Args>
-		static inline void print(const char* format, const Args&... args)
+		static inline void print(const char* format, Args&&... args)
 		{
-			env()->io()->message(MSG_MESSAGEONLY, {}, format, args...);
+			env()->io()->message(
+			    MSG_MESSAGEONLY, {}, format, std::forward<Args>(args)...);
 		}
 
 		template <typename... Args>
-		static inline void info(const char* format, const Args&... args)
+		static inline void info(const char* format, Args&&... args)
```

Comment:
newer versions of fmt raise compilation error when passing the output of "fmt::join" as const lval ref (must be passed as rval)

Reply:
Cool! And this will still work with the older versions then?

Reply:
yes, should work fine. the compile-time check is new, but the implementation is same, so potentially fixes older version as well.
https://github.com/fmtlib/fmt/issues/1225